sdk/program/src/instruction.rs

```diff
@@ -644,3 +645,113 @@ impl CompiledInstruction {
         &program_ids[self.program_id_index as usize]
     }
 }
+
+/// Use to query and convey information about the inner instruction components
+/// when calling the `sol_get_processed_inner_instruction` syscall.
+#[repr(C)]
+#[derive(Default, Debug, Clone, Copy)]
+pub struct InnerMeta {
+    pub data_len: usize,
+    pub accounts_len: usize,
+    pub depth: usize,
+}
+
+/// Returns a inner instruction from the processed inner instruction list.
+///
+/// The processed inner instruction list is a reverse-ordered list of
+/// successfully processed inner instructions. For example, given the call flow:
+///
+/// A
+/// B -> C -> D
+/// B -> E
+/// B -> F
+///
+/// Then B's processed inner instruction list is: [(1, F), (1, E), (1, C), (2, D)]
```

Comment:
@CriesofCarrots Can you double-check that this meets your use case

Reply:
Thanks, @jackcmay , this will work